src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -652,20 +652,29 @@ public ListenableFuture<Iterable<Digest>> findMissingBlobs(
 
     // This is a faster strategy to check missing blobs which does not require querying the CAS.
     // With hundreds of worker machines, it may be too expensive to query all of them for "find
-    // missing blobs".  Assuming the backplane has an accurate accounting of which blobs are hosted
-    // on which workers, the server could instead check the backplane to determine which blobs are
-    // missing.  This may come with risks if the backplane is not up-to-date with the workers or is
-    // not considered an authortiative source of truth.  However, checking workers directly is not a
+    // missing blobs".
+    // Workers register themselves with the backplane for a 30-second window, and if they fail to
+    // re-register within this time frame, they are automatically removed from the backplane. While
+    // this alternative strategy for finding missing blobs is faster and more cost-effective than
+    // the exhaustive approach of querying each worker to find the digest, it comes with a higher
+    // risk of returning expired workers. This is because the strategy may return workers that
+    // have expired in the last 30 seconds. However, checking workers directly is not a
     // guarantee either since workers could leave the cluster after being queried.  Ultimitely, it
     // will come down to the client's resiliency if the backplane is out-of-date and the server lies
     // about which blobs are actually present.  We provide this alternative strategy for calculating
     // missing blobs.
+
     if (configs.getServer().isFindMissingBlobsViaBackplane()) {
       try {
         Map<Digest, Set<String>> foundBlobs = backplane.getBlobDigestsWorkers(blobDigests);
+        Set<String> workerSet = backplane.getStorageWorkers();
         return immediateFuture(
             StreamSupport.stream(blobDigests.spliterator(), false)
-                .filter(digest -> !foundBlobs.containsKey(digest))
+                .filter(
```

Comment:
EOL Comment here: _// best effort to present digests only missing on active workers_